newspaper/article.py

```diff
@@ -160,14 +161,26 @@ def download(self, input_html=None, title=None, recursion_counter=0):
         infinite
         """
         if input_html is None:
-            try:
-                html = network.get_html_2XX_only(self.url, self.config)
-            except requests.exceptions.RequestException as e:
-                self.download_state = ArticleDownloadState.FAILED_RESPONSE
-                self.download_exception_msg = str(e)
-                log.debug('Download failed on URL %s because of %s' %
-                          (self.url, self.download_exception_msg))
-                return
+            parsed_url = urlparse(self.url)
+            if parsed_url.scheme == "file":
+                try:
+                    with open(parsed_url.path, "r") as fin:
+                        html = fin.read()
+                except OSError as e:
+                    self.download_state = ArticleDownloadState.FAILED_RESPONSE
+                    self.download_exception_msg = e.strerror
+                    log.debug('Download failed on URL %s because of %s' %
+                              (self.url, self.download_exception_msg))
+                    return
+            else:
+                try:
+                    html = network.get_html_2XX_only(self.url, self.config)
+                except requests.exceptions.RequestException as e:
+                    self.download_state = ArticleDownloadState.FAILED_RESPONSE
```

Comment:
alternatively maybe we can have a way to share logic for the error handing, since the error handing of both file and http failing are quite similar